src/progress-bar/interfaces.ts

```diff
@@ -69,7 +69,43 @@ export interface ProgressBarProps extends BaseComponentProps {
   onResultButtonClick?: NonCancelableEventHandler;
 }
 
+interface PercentageProgressBarProps extends BaseProgressBarProps {
+  /**
+   * Specifies the progress type.
+   *
+   * @defaultValue 'percentage'
+   */
+  type?: 'percentage';
+}
+
+interface RatioProgressBarProps extends BaseProgressBarProps {
+  /**
+   * Specifies the progress type.
+   *
+   * @defaultValue 'percentage'
+   */
+  type: 'ratio';
+
+  /**
+   * Specifies the maximum value of the progress when type ratio is selected.
+   *
+   * @defaultValue 100
+   */
+  maxValue?: number;
+
+  /**
+   * Control localization of the progress bar value.
+   *
+   * @defaultValue If type === `percentage`, `${value}%`
+   * @defaultValue If type === `ratio`, `${value} of ${maxValue}`
+   */
+  ariaValueText?: string;
+}
+
+export type ProgressBarProps = PercentageProgressBarProps | RatioProgressBarProps;
```

Comment:
I'm not sure if our documenter and typedoc parser can handle this. There's some expectations around how public components define their interfaces, so that we can automatically generate the API docs for the website.

I would suggest we play it safe and use the regular exported interface for both progress bar types